versioning/versions/versions.go

```diff
@@ -31,7 +32,8 @@ type Version struct {
 }
 
 type Spec struct {
-	Versions []Version
+	Versions   []Version
+	Extensions map[string]map[string]string
```

Comment:
OK, I still feel the extension mechanism here unsatisfying. It's hard to connect what defined the YAML file and what the template does.

I have a proposal for this here: https://github.com/GoogleCloudPlatform/runtimes-common/commit/f6ed5287162a8c42d4c2212e51fc179e428043cf

I think the `FROM` feature is simple enough for us to not worry about. My proposal is essentially introducing a function that the template can freely use instead. The template change can be found here: https://github.com/GoogleCloudPlatform/postgresql-docker/commit/f75608afada442cf848ee2ef1a1a83e997a9ed6c